app/models/ems_refresh/save_inventory.rb

```diff
@@ -309,4 +309,33 @@ def save_snapshots_inventory(vm, hashes)
   def save_event_logs_inventory(os, hashes)
     save_inventory_multi(os.event_logs, hashes, :use_association, [:uid])
   end
+
+  def save_new_target(target_hash)
+    unless target_hash[:vm].nil?
+      vm_hash = target_hash[:vm]
+      existing_vm = VmOrTemplate.find_by(:ems_ref => vm_hash[:ems_ref])
```

Comment:
This needs to be scoped by the ems_id.  The ems_ref could be identical (think `vms/1`) in multiple EMSes.